acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,205 @@
+    elif space_output['spaces'].keys()[0] != space_name:
+        raise JujuAssertionError(
+            'Incorrect space name. Found: {}; Expected: {}.'.format(
+            space_output['spaces'].keys()[0], space_name))
+    elif len(space_output['spaces'][space_name].keys()) != 1:
+        raise JujuAssertionError(
+            'Incorrect number of CIDR(s). Found: {}; Expected: 1.'.format(
+            str(len(space_output['spaces'][space_name].keys()))))
+    elif space_output['spaces'][space_name].keys()[0] != subnet_cidr:
+        raise JujuAssertionError(
+            'Incorrect CIDR name. Found: {}; Expected: {}.'.format(
+            space_output['spaces'][space_name].keys()[0], subnet_cidr))
+    else:
+        log.info('New added space {} has been validated.'.format(space_name))
+
+
+def assert_app_status(client, charm_name, expected):
+    """Validate app status after each deployment.
+       :param client: ModelClient object.
+       :param charm_name: Application name for the status check against.
+       :param expected: Expected app status, in this case it is active."""
+    # Run juju status --format json
+    log.info('Checking current status of app {}.'.format(charm_name))
+    status_output = json.loads(
+        client.get_juju_output('status', '--format', 'json', include_e=False))
+    app_status = status_output['applications'][charm_name]['application-status']['current']
+
+    if app_status != expected:
+        raise JujuAssertionError(
+            'App status is incorrect. '\
+            'Found: {}\nExpected: {}\n.'.format(
+            app_status, expected))
+    else:
+        log.info('The current status of app {} is: {}; Expected: {}'.format(
+        charm_name, app_status, expected))
+
+
+def parse_args(argv):
+    """Parse all arguments."""
+    parser = argparse.ArgumentParser(
+        description="Test for lxd spaces and subnets constraints.")
+    add_basic_testing_arguments(parser)
+    return parser.parse_args(argv)
+
+
+def assess_lxd_container_space(client):
+    """Execute full workflow, see docstring on top of the script.
+       :param client: ModelClient object."""
+    assert_initial_spaces(client)
+    assert_initial_subnets(client)
+    assert_added_space(client)
+    # Run juju deploy ubuntu --constraints "spaces=testspace"
+    client.deploy(charm='ubuntu', constraints='spaces=testspace')
+    client.wait_for_started()
+    client.wait_for_workloads()
+    assert_app_status(client, charm_name='ubuntu', expected='active')
+    # Run juju deploy ubuntu ubuntu-lxd --to lxd:0
+    client.deploy(charm='ubuntu', service='ubuntu-lxd', to='lxd:0')
+    client.wait_for_started()
+    client.wait_for_workloads()
+    assert_app_status(client, charm_name='ubuntu-lxd', expected='active')
+
+
+def main(argv=None):
+    args = parse_args(argv)
+    configure_logging(args.verbose)
+    environment = os.getenv('ENV', default='parallel-aws')
+    if environment != 'parallel-aws':
```

Comment:
I understand this is confusing, but Chris's advice is actually correct from before. Jujupy is doing confusing things with environments.yaml and credentials.yaml. I would advocate not throwing a hard stop in here actually, as that's annoying to work around. Throw a warning if you find it's not running on the AWS substrate, but let it continue.